include/libopencm3/stm32/h7/rcc.h

```diff
@@ -181,23 +328,228 @@ LGPL License Terms @ref lgpl_license
+#define RCC_D2CCIP1R_SPDIFSEL_HSI       0x3
+#define RCC_D2CCIP1R_SPI45SEL_APB4      0x0
+#define RCC_D2CCIP1R_SPI45SEL_PLL2Q     0x1
+#define RCC_D2CCIP1R_SPI45SEL_PLL3Q     0x2
+#define RCC_D2CCIP1R_SPI45SEL_HSI       0x3
+#define RCC_D2CCIP1R_SPI45SEL_CSI       0x4
+#define RCC_D2CCIP1R_SPI45SEL_HSE       0x5
+#define RCC_D2CCIP1R_SPI45SEL_MASK      0x7
+#define RCC_D2CCIP1R_SPI123SEL_PLL1Q    0x0
+#define RCC_D2CCIP1R_SPI123SEL_PLL2P    0x1
+#define RCC_D2CCIP1R_SPI123SEL_PLL3P    0x2
+#define RCC_D2CCIP1R_SPI123SEL_I2SCKIN  0x3
+#define RCC_D2CCIP1R_SPI123SEL_PERCK    0x4
+#define RCC_D2CCIP1R_SPI123SEL_MASK     0x7
+#define RCC_D2CCIP1R_SAISEL_PLL1Q       0x0
+#define RCC_D2CCIP1R_SAISEL_PLL2P       0x1
+#define RCC_D2CCIP1R_SAISEL_PLL3P       0x2
+#define RCC_D2CCIP1R_SAISEL_I2SCKIN     0x3
+#define RCC_D2CCIP1R_SAISEL_PERCK       0x4
+#define RCC_D2CCIP1R_SAISEL_MASK        0x7
+/**@}*/
+
+/** @defgroup rcc_d2ccip2r_values RCC_D2CCIP2R Values
+ * @ingroup rcc_registers
+ * @{*/
+#define RCC_D2CCIP2R_LPTIM1SEL_SHIFT        28
+#define RCC_D2CCIP2R_CECSEL_SHIFT           22
+#define RCC_D2CCIP2R_USBSEL_SHIFT           20
+#define RCC_D2CCIP2R_I2C123SEL_SHIFT        12
+#define RCC_D2CCIP2R_RNGSEL_SHIFT           8
+#define RCC_D2CCIP2R_USART16SEL_SHIFT       3
+#define RCC_D2CCIP2R_USART234578SEL_SHIFT   0
+
+#define RCC_D2CCIP2R_USART16SEL_PCLK2       0
+#define RCC_D2CCIP2R_USART234578SEL_PCLK1   0
+#define RCC_D2CCIP2R_USARTSEL_PLL2Q         1
+#define RCC_D2CCIP2R_USARTSEL_PLL3Q         2
+#define RCC_D2CCIP2R_USARTSEL_HSI           3
+#define RCC_D2CCIP2R_USARTSEL_CSI           4
+#define RCC_D2CCIP2R_USARTSEL_LSE           5
+#define RCC_D2CCIP2R_USARTSEL_MASK          7
+/**@}*/
+
+
+#define HSI_BASE_FREQUENCY              64000000UL
+
+/** Enumerations for clocks in the clock tree to allow user to get the current configuration of the
+ *  clocks from the RCC module. These clock sources will each be tracked through the settings.
+ */
+enum rcc_clock_source {
+  RCC_CPUCLK,
+  RCC_SYSCLK,
+  RCC_PERCLK,
+  RCC_SYSTICKCLK,
+  RCC_HCLK3,
+  RCC_AHBCLK,           /* AHB1,2,4 all share base HCLK. */
+  RCC_APB1CLK,          /* APB1 and PCLK1 are aliases. */
+  RCC_PCLK1,
+  RCC_APB2CLK,
+  RCC_PCLK2,
+  RCC_APB3CLK,
+  RCC_PCLK3,
+  RCC_APB4CLK,
+  RCC_PCLK4,
+  RCC_FDCAN1CLK,        /* FDCAN controllers share a kernel clock */
+  RCC_FDCAN2CLK,
+  RCC_SPI1CLK,          /* SPI 1/2/3 share a kernel clock. */
+  RCC_SPI2CLK,
+  RCC_SPI3CLK,
+  RCC_SPI4CLK,          /* SPI 4/5 share kernel clock. */
+  RCC_SPI5CLK,
+  RCC_SPI6CLK,          /* SPI6 has its own kernel clock. */
+  RCC_USART1CLK,        /* USART 1/6 share kernel clock. */
+  RCC_USART6CLK,
+  RCC_USART2CLK,        /* USART 2/3/4/5/6/8 share kernel clock. */
+  RCC_USART3CLK,
+  RCC_USART4CLK,
+  RCC_USART5CLK,
+  RCC_USART7CLK,
+  RCC_USART8CLK,
+};
 
 enum rcc_osc {
   RCC_PLL,
   RCC_HSE,
   RCC_HSI,
-  RCC_LSE,
-  RCC_LSI
+  RCC_LSE
```

Comment:
Why are we dropping LSI here again?

Reply:
Mainly for simplicity...I didn't think it useful to run off an internal 32KHz clock anytime in the near future, and it requires managing a bit more state.

Reply:
well, you're slavishly holding all options elsewhere, and then not here?  I agree, running off lsi is pointless, but it still seems odd to just drop it.